AEPNotificationContent/Sources/Extensions/UNNotificationContent+TemplateKeys.swift

```diff
@@ -0,0 +1,156 @@
+    // If the mode is not available, defaults to auto
+    var carouselMode: CarouselMode {
+        guard let modeString = userInfo[AEPNotificationContentConstants.PayloadKey.Carousel.MODE] as? String,
+              let mode = CarouselMode(rawValue: modeString) else {
+            return .auto
+        }
+        return mode
+    }
+
+    // MARK: - Timer Template Keys
+
+    // Returns the alternate title of timer template
+    // If the alternate title is not available, nil is returned
+    var alternateTitle: String? {
+        userInfo[AEPNotificationContentConstants.PayloadKey.Timer.ALTERNATE_TITLE] as? String
+    }
+
+    // Returns the alternate body of timer template
+    // If the alternate body is not available, nil is returned
+    var alternateBody: String? {
+        userInfo[AEPNotificationContentConstants.PayloadKey.Timer.ALTERNATE_BODY] as? String
+    }
+
+    // Returns the alternate image URL of timer template
+    // If a valid alternate image URL is not available, nil is returned
+    var alternateImageURL: URL? {
+        guard let alternateImageURLString = userInfo[AEPNotificationContentConstants.PayloadKey.Timer.ALTERNATE_IMAGE] as? String else {
+            return nil
+        }
+        return URL(string: alternateImageURLString)
+    }
+
+    // Return the timer color from the notification
+    // If a valid timer color is not available, defaultTitle color is used
+    var timerColor: UIColor {
+        UNNotificationContent.extractColor(from: self, key: AEPNotificationContentConstants.PayloadKey.Timer.COLOR,
+                                           defaultColor: .defaultTitle)
+    }
+
+    // Returns the duration of the timer
+    // If the duration is not available, nil is returned
+    var timerDuration: TimeInterval? {
+        guard let durationString = userInfo[AEPNotificationContentConstants.PayloadKey.Timer.DURATION] as? String,
+              let duration = TimeInterval(durationString) else {
+            return nil
+        }
+        return duration
+    }
+
+    // Returns the end timestamp of the timer
+    // If the end timestamp is not available, nil is returned
+    var endTimestamp: TimeInterval? {
+        guard let endTimestampString = userInfo[AEPNotificationContentConstants.PayloadKey.Timer.END_TIMESTAMP] as? String,
+              let endTimestamp = TimeInterval(endTimestampString) else {
+            return nil
+        }
+        return endTimestamp
+    }
+
+    /// Extracts and returns a color from notification content for a specified key.
+    /// - Parameters:
+    ///   - notificationContent: The notification content to extract from.
+    ///   - key: The key for the color information.
+    ///   - defaultColor: The default color to return if extraction fails.
+    /// - Returns: The extracted color or the default color.
+    static func extractColor(from notificationContent: UNNotificationContent, key: String, defaultColor: UIColor) -> UIColor {
```

Comment:
any reason this is static instead of just an instance method?

Reply:
it should also be private unless you want other files using it :)

Reply:
it should also be private unless you want other files using it :)

Reply:
100% This can be private and non-static.